Add support for rollup@4

Hi and thanks for this plugin

rollup-plugin-sizes@1.0.5 declares `"rollup": "^2 || ^3"` in peerDependencies

Running npm install in a projet using rollup@4 generates the following error
```
 npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: rollup-plugin-sizes@1.0.5
npm ERR! Found: rollup@4.0.2
npm ERR! node_modules/rollup
npm ERR!   dev rollup@"~4.0.2" from the root project
npm ERR!   peerOptional rollup@"^2.68.0||^3.0.0||^4.0.0" from @rollup/plugin-commonjs@25.0.5
npm ERR!   node_modules/@rollup/plugin-commonjs
npm ERR!     dev @rollup/plugin-commonjs@"~25.0.5" from the root project
npm ERR!   5 more (@rollup/plugin-node-resolve, @rollup/plugin-terser, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer rollup@"^2 || ^3" from rollup-plugin-sizes@1.0.5
npm ERR! node_modules/rollup-plugin-sizes
npm ERR!   dev rollup-plugin-sizes@"~1.0.5" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: rollup@3.29.4
npm ERR! node_modules/rollup
npm ERR!   peer rollup@"^2 || ^3" from rollup-plugin-sizes@1.0.5
npm ERR!   node_modules/rollup-plugin-sizes
npm ERR!     dev rollup-plugin-sizes@"~1.0.5" from the root project
```

Possible workaround that could be evaluated (same as in #142): use the `--legacy-peer-deps` option

--- a/.changeset/long-geckos-perform.md
+++ b/.changeset/long-geckos-perform.md
@@ -1,0 +1,5 @@
+---
+"rollup-plugin-sizes": patch
+---
+
+feat: add rollup@4 support

--- a/package.json
+++ b/package.json
@@ -33,6 +33,6 @@
     "module-details-from-path": "^1.0.3"
   },
   "peerDependencies": {
-    "rollup": "^2 || ^3"
+    "rollup": "^2 || ^3 || ^4"
   }
 }

